ocaml/xapi/xapi_pool.ml

```diff
@@ -615,6 +615,44 @@ let protect_exn f x =
 		Debug.log_backtrace e (Backtrace.get e);
 		None
 
+let create_or_get_pvs_server_on_master __context rpc session_id pvs_site_ref (pvs_server_ref, pvs_server) : API.ref_PVS_server =
+	let my_uuid = pvs_server.API.pVS_server_uuid in
+	let new_pvs_server_ref =
+		try Client.PVS_server.get_by_uuid ~rpc ~session_id ~uuid:my_uuid
+		with _ ->
+			debug "Found no PVS server with uuid = '%s' on the master, so creating one." my_uuid;
+			Client.PVS_server.introduce ~rpc ~session_id
+			~addresses:pvs_server.API.pVS_server_addresses
+			~first_port:pvs_server.API.pVS_server_first_port
+			~last_port:pvs_server.API.pVS_server_last_port
+			~site:pvs_site_ref
+	in
+	new_pvs_server_ref
+
+let create_or_get_pvs_site_on_master __context rpc session_id (pvs_site_ref, pvs_site) : API.ref_PVS_site =
+	let my_site_name = pvs_site.API.pVS_site_name in
+	let new_pvs_site_ref =
+		let expr = "field \"name\"=\"" ^ my_site_name ^ "\"" in
```

Comment:
Its taking String for this case.
https://github.com/xapi-project/xen-api/blob/master/ocaml/idl/datamodel.ml#L8205
There are places we have used it https://github.com/xapi-project/xen-api/blob/b91c8e8d244d3967ae33eb0f52501ceafca7a64c/ocaml/xapi/import.ml#L793